clang/lib/Parse/ParseCXXInlineMethods.cpp

```diff
@@ -1161,6 +1161,12 @@ bool Parser::ConsumeAndStoreInitializer(CachedTokens &Toks,
 
   while (true) {
     switch (Tok.getKind()) {
+    case tok::ellipsis:
+      // We found an elipsis at the end of the parameter list;
+      // it is not part of a parameter declaration.
+      if (ParenCount == 1 && NextToken().is(tok::r_paren))
```

Comment:
Using `ParenCount` here looks wrong to me -- the entire class could be in parens, as could the function declarator. Do these work:

```c++
struct A {
    void (f(int = 0 ...));
};

void f() {
    ([] {
        struct B {
            void f(int = 0 ...);
        };
    });
}
```

I think checking `ParenCount` is also redundant; if we see a `(`, we skip straight to the matching `)`, so any token we see here is always at the top level anyway.